framework/src/main/java/org/apache/drill/test/framework/TestDriver.java

```diff
@@ -268,16 +274,123 @@ public int runTests() throws Exception {
       totalExecutionFailure += executionFailures.size();
       totalVerificationFailure += verificationFailures.size();
       totalTimeoutFailure += timeoutFailures.size();
-      totalCanceledTest += canceledTests.size(); 
+      totalCanceledTest += canceledTests.size();
+      totalFailingTest = 0;
+      totalFailingTest = totalExecutionFailure + totalVerificationFailure + totalTimeoutFailure + totalCanceledTest;
+      failedTests.addAll(executionFailures);
+      failedTests.addAll(verificationFailures);
+      failedTests.addAll(timeoutFailures);
+      failedTests.addAll(canceledTests);
+ 
     }
+    
+     if (totalFailingTest/(totalPassingTest +totalFailingTest) <= cmdParam.percentThresholdForRandomFailure && totalFailingTest != 0){
+      LOG.info("Found "+totalFailingTest+" errors again running them to confirm");
```

Comment:
Change to "Found totalFailingTest errors. Running the failed tests again to isolate random failures"